src/Connectivity/pointer_jumping.jl

```diff
@@ -0,0 +1,47 @@
+"""
+    struct PointerJumping <: WeakConnectivityAlgorithm
+
+A struct representing a [`WeakConnectivityAlgorithm`](@ref)
+based on parallel opportunistic pointer jumping
+
+### References
+- Connected-Components algorithms for Mesh-Connected Parallel Computers
+Goddard, Kumar and Prins.
+http://cse.unl.edu/~goddard/Papers/Journals/dimacs.pdf
+"""
+struct PointerJumping <: WeakConnectivityAlgorithm end
+
+function connected_components(g::SimpleGraph{T}, alg::PointerJumping) where T <: Integer
```

Comment:
Why is this not `AbstractGraph{T}`?

If it works with `<:AbstractGraph` then you can remove the `SimpleGraph` import in `Connectivity.jl`.